api/v1alpha1/paasconfig_types.go

```diff
@@ -35,11 +35,14 @@ const (
 // +kubebuilder:subresource:status
 // +kubebuilder:resource:path=paasconfig,scope=Cluster
 type PaasConfig struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
+	metav1.TypeMeta `json:",inline"`
+	// +kubebuilder:validation:Optional
+	metav1.ObjectMeta `json:"metadata"`
 
-	Spec   PaasConfigSpec   `json:"spec,omitempty"`
-	Status PaasConfigStatus `json:"status,omitempty"`
+	// +kubebuilder:validation:Optional
+	Spec PaasConfigSpec `json:"spec"`
+	// +kubebuilder:validation:Optional
```

Comment:
As represented in the generated manifest, this has no effect nor is needed. Please don't add this here.

Reply:
For consistency, I have reverted all // +kubebuilder:validation:Optional tags to omitempty flags.